Usual patch to produce Linux-like .so files on FreeBSD

--- a/ltmain.sh	Wed Aug 29 14:28:46 2007
+++ b/ltmain.sh	Wed Jun 25 13:06:05 2008
@@ -6195,9 +6195,9 @@
 	    revision="$number_revision"
 	    ;;
 	  freebsd-aout|freebsd-elf|qnx|sunos)
-	    current="$number_major"
-	    revision="$number_minor"
-	    age="0"
+	    current=`expr $number_major + $number_minor`
+	    age="$number_minor"
+	    revision="$number_revision"
 	    ;;
 	  irix|nonstopux)
 	    func_arith $number_major + $number_minor
@@ -6271,8 +6271,8 @@
 	  ;;
 
 	freebsd-elf)
-	  major=".$current"
-	  versuffix=".$current"
+	  major=.`expr $current - $age`
+	  versuffix="$major.$age.$revision"
 	  ;;
 
 	irix | nonstopux)
